### PR TITLE
added inputSearchModel attribute exposing input search value

### PR DESCRIPTION
**Pull request by abner**
added inputSearchModel attribute exposing input search value to a model in the parent scope
Solves Issue #829

### CODE REVIEW

**Comment by brunobastosg** [comment]
+1. Exactly what I needed.

**Comment by abner** [comment]
Hi. it is possible to have this pull request accepted? in our company we have a project using ui-sellect and we wish to use this feature

**Comment by hellmanj** [comment]
+1 Please merge!

File: test/select.spec.js [hunk 1]
```diff
@@ -1904,4 +1904,46 @@ describe('ui-select tests', function() {
     });
   });
 
+  describe('expose input search value to a model in the parent scope', function() {
+
+        var el;
```
Comment (wesleycho) [inline test/select.spec.js]: Indentation is off.

File: src/uiSelectDirective.js [hunk 2]
```diff
@@ -68,6 +68,18 @@ uis.directive('uiSelect',
...
 
+        if(angular.isDefined(attrs.inputSearchModel)){
+          $select.search = scope.$parent.$eval(attrs.inputSearchModel);
+
+          scope.$parent.$watch(attrs.inputSearchModel, function(){
+            $select.search = scope.$parent.$eval(attrs.inputSearchModel);
+          });
+
+          scope.$watch('$select.search', function(){
+             scope.$parent.$eval(attrs.inputSearchModel + '= "' + $select.search + '"');
```
Comment (wesleycho) [inline src/uiSelectDirective.js]: A better approach is above before the $watch, cache via `var foo = $parse(attrs.inputSearchModel + '= "' + $select.search + '"')` and inside the watch listener callback apply scope.$parent, i.e. `foo(scope.$parent)`, with obviously better variable names.

**Comment by user378230** [comment]
Fairly simple to do with a custom directive...

```js
myModule.directive('myUiSelect', function() {
  return {
    require: 'uiSelect',
    link: function(scope, element, attrs, $select) {
      scope.$watch(function() { return $select.search; }, function(newSearchValue) {
        //...
      });
    }
  };
});
```
You can also then programmatically set `$select.search` to a new value if you wish.

(Ref: [7. Accessing `$select`](https://github.com/angular-ui/ui-select/wiki/Accessing-$select))

**Pull request closed without merging**